client/src/components/molecules/NavList/index.tsx

```diff
@@ -31,162 +48,164 @@ const NavList: FC<Props> = ({ lists, isOpenSidebar }): JSX.Element => {
     setOpenIndexes((prevOpenIndexes: number[]) => {
       if (prevOpenIndexes.includes(index)) {
         return prevOpenIndexes.filter((i: number) => i !== index)
-      } else {
-        return [...prevOpenIndexes, index]
       }
+      return [...prevOpenIndexes, index]
     })
   }
 
+  if (isLoading) {
+    return <SkeletonNavList {...{ isOpenSidebar }} />
+  }
+
   return (
     <>
-      {!isLoading ? (
-        <>
-          {lists.map((item, index) => (
-            <div key={index}>
-              {item.role?.map((r) => r).includes(user?.userById.role.name as string) ? (
-                <Tooltip
-                  placement="right"
-                  overlay={item.name}
-                  overlayClassName={classNames(isOpenSidebar ? '!opacity-0' : '!opacity-100')}
-                  arrowContent={<div className="rc-tooltip-arrow-inner"></div>}
-                >
-                  <div>
-                    {item.submenu === true && isOpenSidebar ? (
-                      <div
-                        onClick={() => toggleSubmenu(index)}
-                        className={classNames(
-                          'relative flex cursor-pointer items-center transition',
-                          'duration-75 ease-in-out hover:bg-slate-100 hover:text-slate-700'
-                        )}
+      {lists.map((item, index) => {
+        const isRoleAllowed = item.role?.includes(user?.userById?.role?.name as string)
+        if (!isRoleAllowed) return null
+
+        const isAdminRole = user?.userById?.role?.name === Roles.HR_ADMIN
+
+        // If current menu is Leave Management and user is LEAD (but not HR_ADMIN),
```

Comment:
Please review other comments and rephrase it
```suggestion
        // Check if user role is LEAD and show Leader List submenu items only
```